rlwe/decryptor.go

```diff
@@ -0,0 +1,109 @@
+package rlwe
+
+import (
+	"github.com/ldsec/lattigo/v2/ring"
+	"github.com/ldsec/lattigo/v2/utils"
+)
+
+// Decryptor is an interface for decrypting Ciphertexts. A Decryptor stores the secret-key.
+type Decryptor interface {
+	// DecryptNew decrypts the ciphertext and returns a newly created
+	// plaintext. A Horner method is used for evaluating the decryption.
+	// The level of the output plaintext is ciphertext.Level().
+	DecryptNew(ciphertext *Element) (plaintext *Plaintext)
+
+	// DecryptNew decrypts the ciphertext and returns a newly created
+	// plaintext. A Horner method is used for evaluating the decryption.
+	// The level of the output plaintext is ciphertext.Level().
+	// Plaintext output is in the NTT domain.
+	DecryptNTTNew(ciphertext *Element) (plaintext *Plaintext)
```

Comment:
This is only for CKKS right ? Wouldn't it be preferable to simply have a final NTT in the `ckks.Decryptor.Decrypt` `?

Reply:
Because the plaintext is created by the decryptor, it must be told what the output domain of the plaintext is.

Reply:
Remove [...]New methods from RLWE and wrap inplace method in schemes

Reply:
Done

Reply:
done